src/core_plugins/kibana/public/context/app.js

```diff
@@ -0,0 +1,100 @@
+import _ from 'lodash';
+
+import uiModules from 'ui/modules';
+import contextAppTemplate from './app.html';
+import './components/loading_button';
+import './components/size_picker/size_picker';
+import {
+  createInitialQueryParametersState,
+  QueryParameterActionsProvider,
+  QUERY_PARAMETER_KEYS,
+} from './query_parameters';
+import { QueryActionsProvider } from './query';
+
+const module = uiModules.get('apps/context', [
+  'elasticsearch',
+  'kibana',
+  'kibana/config',
+  'kibana/notify',
+  'ngRoute',
+]);
+
+module.directive('contextApp', function ContextApp() {
+  return {
+    bindToController: true,
+    controller: ContextAppController,
+    controllerAs: 'contextApp',
+    restrict: 'E',
+    scope: {
+      anchorUid: '=',
+      columns: '=',
+      indexPattern: '=',
+      predecessorCount: '=',
+      successorCount: '=',
+      sort: '=',
+      discoverUrl: '=',
+    },
+    template: contextAppTemplate,
+  };
+});
+
+function ContextAppController($scope, Private) {
+  const queryParameterActions = Private(QueryParameterActionsProvider);
+  const queryActions = Private(QueryActionsProvider);
+
+  this.state = createInitialState(this.discoverUrl);
+
+  this.actions = _.mapValues(Object.assign(
+    {},
+    queryParameterActions,
+    queryActions,
+  ), (action) => (...args) => action(this.state)(...args));
```

Comment:
Yes, this is OK with me, though I still would prefer "dumber" code like I suggested, because it's much simpler.

The one benefit I can see to your suggestion is that it prevents someone from accidentally assigning the wrong method to an action, which is more likely in my suggestion. Personally, I don't think we're likely to encounter this problem in general. I'm a big believer in "code is written once and read many times", so I'd rather write it in a way that makes it very clear what's going on, even if it means some verbosity.

That said, though, it's your call! :)